serenity-core/src/test/java/net/thucydides/core/reports/remoteTesting/WhenGeneratingABrowserStackLink.java

```diff
@@ -0,0 +1,76 @@
+package net.thucydides.core.reports.remoteTesting;
+
+import net.thucydides.core.model.TestOutcome;
+import net.thucydides.core.util.EnvironmentVariables;
+import net.thucydides.core.util.MockEnvironmentVariables;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.mockito.Mockito.when;
+
+/**
+ * Browserstack tests for link generation.
+ * User: johnsmart, bebef1987
+ * Date: 8/03/12
+ * Time: 8:18 AM
+ */
+public class WhenGeneratingABrowserStackLink {
+
+    EnvironmentVariables environmentVariables;
+
+    private final String A_SESSION_ID = "5f9fef27854ca50a3c132ce331cb6034";
+
+    @Mock
+    TestOutcome testOutcome;
+
+    @Mock
+    TestOutcome testOutcomeWithNoSessionId;
+
+    @Before
+    public void initMocks() {
+
+        MockitoAnnotations.initMocks(this);
+        environmentVariables = new MockEnvironmentVariables();
+        when(testOutcome.getSessionId()).thenReturn(A_SESSION_ID);
+    }
+
+    @Test
+    public void should_generate_no_link_if_no_user_anme_is_provided() {
+        environmentVariables.setProperty("browserstack.url","http://:API_key@hub.browserstack.com:80/wd/hub");
+        RemoteTestingLinkManager remoteTestingLinkManager = new RemoteTestingLinkManager(environmentVariables);
+
+        String browserStackLink = remoteTestingLinkManager.linkFor(testOutcome);
+        assertThat(browserStackLink, is(nullValue()));
+    }
+
+    @Test
```

Comment:
Could you add a positive test as well to demonstrate building the BrowserStack video url?